#
# Adapted from synth-explorer/tools/ghdl-wasm/grt-stdio.patch.
# Copyright 2025 synth-explorer contributors.
# Licensed under the Apache License, Version 2.0.
# https://www.apache.org/licenses/LICENSE-2.0
# SPDX-License-Identifier: Apache-2.0
#
--- grt-stdio.ads
+++ grt-stdio.ads
@@ -86,0 +87,4 @@
+
+   function C_Fputc (C : int; Stream : FILEs) return int;
+   function C_Fflush (Stream : FILEs) return int;
+   function C_Fclose (Stream : FILEs) return int;
@@ -94 +98 @@
-   pragma Import (C, fputc, "putc_unlocked");
+   pragma Import (C, C_Fputc, "putc_unlocked");
@@ -100 +104 @@
-   pragma Import (C, fflush);
+   pragma Import (C, C_Fflush, "fflush");
@@ -103 +107 @@
-   pragma Import (C, fclose);
+   pragma Import (C, C_Fclose, "fclose");
